### PR TITLE
add API: GET default config

### PR DESCRIPTION
**Pull request by g-back**
follow up: allow to SET default config via API?

### CODE REVIEW

**Comment by larshp** [comment]
each time an object is checked, the abaplint configuration for that check is sent to the API from ABAP

**Comment by g-back** [comment]
Yea - but use case may be this:
- set a central default config
- check objects with this central config instead of using the package-specific abaplint.json

**Comment by larshp** [comment]
thats done in the ABAP stack, via https://github.com/abaplint/abaplint-abap-backend

image

**Comment by g-back** [comment]
I think I worded it poorly: imagine you have 50 packages in there and you want to change the configuration for all of them (let's say disable a single rule) - you would have to copy paste it 50 times instead of changing it in one place.

**Comment by larshp** [comment]
the code will continue up in the package hierarchy until it hits a configuration, so if there is a top package for the 50 packages, all configuration can be done there

**Comment by g-back** [comment]
It may be that there is no top package - either way, was just a suggestion.

this PR is for abaplint/abaplint-abap-backend#9

File: src/api/index.ts [hunk 1]
```diff
@@ -17,6 +17,10 @@ router.get("/ping", (_req, res) => {
   res.json(createSuccessStringResponse("abap is forevah!"));
 });
 
+router.get("/get_default_configuration", (_,res) => {
+  res.json(getDefaultConfig());
```
Comment (sbcgua) [inline src/api/index.ts:21]: Maybe createSuccessResponse<...>(getDefaultConfig()) to keep the api shape?

**Comment by sbcgua** [comment]
Some thoughts regarding common config:
1) Supposing that abaplint backend would be most probably private, we can store configs as key/value. Where key would be repo url (which is uniq enough) and value - the config. The assignment can be done from UI. So adding a new package at UI side would ping server side to check presence of the config and submit it if missing or changed
2) completely different idea. To implement sessions. On SCI start, it requests (or generates UUID) a session key, and then server can store the cache for some period of time. Including config and dependencies ! Does SCI keeps same abap session between object to have singletons ? This can be a bit tricky but totally implementable

File: src/api/index.ts (hunk 1 above)
Comment (g-back) [reply src/api/index.ts]: Changed it - let me know if that's how you meant it

Comment (sbcgua) [reply src/api/index.ts]: Yeap, this look fine. Would you mind to make a short unit test for this as well ? Example is in https://github.com/abaplint/abaplint-cloud-foundry/blob/master/test/check-object.api.test.ts

No need to check the config of course, just it's presence like in check-object test. With `toMatchObject`

Comment (g-back) [reply src/api/index.ts]: Added test case for API and moved config stuff to new file

**Comment by larshp** [comment]
thanks both 👍

**Pull request merged**